Code/ForceField/MMFF/Params.cpp

```diff
@@ -55,25 +69,34 @@ MMFFAromCollection::MMFFAromCollection(const boost::uint8_t *mmffArom) {
 const boost::uint8_t defaultMMFFArom[] = {37, 38, 39, 44, 58, 59, 63, 64, 65,
                                           66, 69, 76, 78, 79, 80, 81, 82};
 
-class MMFFDefCollection *MMFFDefCollection::ds_instance = NULL;
+boost::scoped_ptr<MMFFDefCollection> MMFFDefCollection::ds_instance;
+#ifdef BOOST_THREAD_PROVIDES_ONCE_CXX11
+boost::once_flag MMFFDefCollection::ds_flag;
+#else
+boost::once_flag MMFFDefCollection::ds_flag = BOOST_ONCE_INIT;
+#endif
 
 extern const std::string defaultMMFFDef;
 
 MMFFDefCollection *MMFFDefCollection::getMMFFDef(const std::string &mmffDef) {
-  if (!ds_instance) {
-    ds_instance = new MMFFDefCollection(mmffDef);
-  } else if (mmffDef != "") {
-    delete ds_instance;
-    ds_instance = new MMFFDefCollection(mmffDef);
+#ifdef RDK_THREADSAFE_SSS
+  boost::call_once(ds_flag, boost::bind(&create, mmffDef));
```

Comment:
This is not the exact same functionality as before, and therefore maybe a bug?

The problem is if 'mmffDef' changes during the lifetime of the program, the MMFFDefCollection won't be re-initialized with the new parameters. This could cause the following subtle race condition: two threads each call getMMFFDef with two different 'mmffDef' parameters; its non-deterministic which of the two definitions will be returned to the two threads. The only deterministic thing is that they will both get the _same_ MMFFDefCollection pointer. 

Since 'default' initialization is likely the behavior we want to make fast, we could do this: 

```cpp
if (mmffDef.empty()) {
  boost::call_once(ds_flag, boost::bind(&create, ""));
  return ds_instance.get();
}
return new MMFFDefCollection(mmffDef);
```

Though the 2nd return I believe then becomes a memory leak because the caller isn't expecting to take ownership. So likely need a better way.  

I think the proper solution is a flyweight that depends on the parameters passed in. But that's probably more expensive.